fixed Validate Model Issues

clefspear opened the pull request.

clefspear committed:
fixed Validate Model Issues

clefspear committed:
test2 recent results

clefspear committed:
update

GabeSantini commented on src/__init__.py:
```diff
@@ -304,6 +304,8 @@ def validate_model(self, test_data=None, label_values=None,
         Returns:
             Dictionary of validation results
         """
+        from .dataProcessing import to_serializable
```
Imports should happen outside of the class

GabeSantini commented on src/__init__.py:
```diff
@@ -317,40 +319,27 @@ def validate_model(self, test_data=None, label_values=None,
         
-        X_test = self.test_data[0]
-        y_test = self.test_data[1].tolist()
+        X_test = test_data['X']
```
This is redundant, modify your check, there is no need to recreate a dictionary when you can access them directly from parameters

GabeSantini commented on src/__init__.py:
```diff
@@ -317,40 +319,27 @@ def validate_model(self, test_data=None, label_values=None,
+        X_test = test_data['X']
+        y_test = test_data['y']
+        
+        # Convert to list if needed
```
another possible redundancy, if you need datasets formated in a specific way, modify processing and make sure model inst impacted.

GabeSantini commented on src/__init__.py:
```diff
@@ -363,31 +352,187 @@ def validate_model(self, test_data=None, label_values=None,
             'y_pred': to_serializable(y_pred),
-            'label_values': label_values
+            'label_values': label_values,
+            'thresholds': {
```
no need to store thresholds, they are stored at parameter level

GabeSantini commented on src/__init__.py:
```diff
@@ -363,31 +352,187 @@ def validate_model(self, test_data=None, label_values=None,
+        
+        # Check thresholds and update passed_thresholds flag
+        try:
+            self.check_thresholds(results, accuracy_t, precision_t, recall_t, f1_t)
```
stop passing attribute as parameters information locally within function, access them directly within the function.

GabeSantini commented on src/__init__.py:
```diff
@@ -363,31 +352,187 @@ def validate_model(self, test_data=None, label_values=None,
+        Raises:
+            ValueError: If any metric falls below its threshold
+        """
+        accuracy = results['accuracy']
```
Redundant, no need to reasign to variable, access directly as attribute

GabeSantini commented on src/__init__.py:
```diff
@@ -485,7 +630,52 @@ def saveModel(self, path: str, name: str):
          print(f'getting model in {path}')
          self.model = keras.models.load_model(path)
+
+
```
if you are using copilot or gpt, make sure to clean up un needed comments

GabeSantini commented on src/__init__.py:
```diff
@@ -363,31 +352,187 @@ def validate_model(self, test_data=None, label_values=None,
+            error_msg = "Validation metrics below thresholds:\n  - " + "\n  - ".join(failed_metrics)
+            raise ValueError(error_msg)
+    
+    def get_validation_summary(self):
```
Not used, also, validation data is being run in multiple machines. thus when comparing multiple runs it should be gathering data from either a log file or a series of log files. Callbacks have been configured for both log files and separately saved as Json files with timestamps.

I suggest not adding this in the class but rather creating a separate function within the modelEvaluation module.

GabeSantini commented on src/__init__.py:
```diff
@@ -363,31 +352,187 @@ def validate_model(self, test_data=None, label_values=None,
+        df = pd.DataFrame(summary_data)
+        return df
+    
+    def validate_with_multiple_thresholds(self, test_data=None, label_values=None,
```
same as above, unused and you are hardcoding values.

in general, model class should house data pertinenet to 1 model run. analysis of multiple models should be independent of class instance. (unless you are loading model history but that is overkill)

GabeSantini reviewed (changes requested):
Before addressing changes, pull from dev, Genjie posted changes.

resolve conflicts before re-submitting pull request

clefspear committed:
- Move dataProcessing import to top of file
- Remove redundant dictionary creation in validate_model()
- Remove threshold storage from results dict (stored as instance attributes)
- Refactor check_thresholds() to access attributes directly without parameters
- Remove get_validation_summary() and validate_with_multiple_thresholds() methods

clefspear commented:
- Move dataProcessing import to top of file
- Remove redundant dictionary creation in validate_model()
- Remove threshold storage from results dict (stored as instance attributes)
- Refactor check_thresholds() to access attributes directly without parameters
- Remove get_validation_summary() and validate_with_multiple_thresholds() methods

clefspear committed:
update test2 results

GabeSantini reviewed (changes requested):
pull from dev first

clefspear committed:
Merge Dev branch and implement PR #34 feedback

clefspear committed:
final edits
